code/modules/economy/quikpay.dm

```diff
@@ -0,0 +1,182 @@
+
+/obj/item/device/nanoquikpay
+	name = "\improper NT Quik-Pay"
+	desc = "Swipe you're ID to make direct company purcheses"
+	icon = 'icons/obj/device.dmi'
+	icon_state = "quikpay"
+	var/machine_id = ""
+	var/list/items = list()
+	var/sum = 0
+	var/access_code = 0
+	var/editmode = 0
+
+
+
+
+/obj/item/device/nanoquikpay/New()
+	..()
+	machine_id = "[station_name()] Quik-Pay #[SSeconomy.num_financial_terminals++]"
+	access_code = rand(1111,111111)
+	spawn(0)
+		print_reference()
+
+		//create a short manual as well
+		var/obj/item/weapon/paper/R = new(src.loc)
+		R.name = "Quik And Easy: How to make a transaction"
+
+		R.info += "<b>Quik-Pay setup:</b><br>"
+		R.info += "1.Remember your access code included on the paper that is included with your device<br>"
+		R.info += "2. Enter the pin into the locking mechanism to be able to add items to the menu<br><br>"
+		R.info += "3. Add items to the menu by typing the item name and its price<br><br>"
+		R.info += "<b>When starting a new transaction:</b><br>"
+		R.info += "1.Have the customer enter the amount of the item they want.<br>"
+		R.info += "2. Press confirm payment.<br>"
+		R.info += "3. Allow them to review the sum.<br>"
+		R.info += "4. Have them swipe their card to pay for the items, press return to reset the sum..<br>"
+
+
+		//stamp the paper
+		var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+		stampoverlay.icon_state = "paper_stamp-cent"
+		if(!R.stamped)
+			R.stamped = new
+		R.offset_x += 0
+		R.offset_y += 0
+		R.ico += "paper_stamp-cent"
+		R.stamped += /obj/item/weapon/stamp
+		R.add_overlay(stampoverlay)
+		R.stamps += "<HR><i>This paper has been stamped by the Head of Personnel's desk.</i>"
+
+
+/obj/item/device/nanoquikpay/proc/print_reference()
+	var/obj/item/weapon/paper/R = new(src.loc)
+	var/pname = "Reference: [machine_id]"
+	var/info = "<b>[machine_id] reference</b><br><br>"
+	info += "Access code: [access_code]<br><br>"
+	info += "<b>Do not lose or misplace this code.</b><br>"
+	R.set_content_unsafe(pname, info)
+
+	//stamp the paper
+	var/image/stampoverlay = image('icons/obj/bureaucracy.dmi')
+	stampoverlay.icon_state = "paper_stamp-cent"
+	if(!R.stamped)
+		R.stamped = new
+	R.stamped += /obj/item/weapon/stamp
+	R.add_overlay(stampoverlay)
+	R.stamps += "<HR><i>This paper has been stamped by the Head of Personnel's desk.</i>"
+	var/obj/item/smallDelivery/D = new(R.loc)
+	R.forceMove(D)
+	D.wrapped = R
+	D.name = "small parcel - 'Quik Pay access code'"
+
+
+
+/obj/item/device/nanoquikpay/attackby(obj/O, mob/user)
+	var/obj/item/weapon/card/id/I = O.GetID()
+	if(!istype(O, /obj/item/weapon/card/id))
```

Comment:
do a
```dm
	if (!istype(O))
		return

	var/datum/money_account/quickpay_account = SSeconomy.get_department_account("Civilian")
	// ... so on
```

There is no need for the empty if clause and the else clause.

Reply:
```suggestion
	if (!istype(O))
		return
```